pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -64,6 +64,23 @@ func (pdev podDevices) delete(pods []string) {
 	}
 }
 
+// Returns list of device Ids allocated to the given pod for the given resource.
+// Returns nil if we don't have cached state for the given <podUID, resource>.
+func (pdev podDevices) podDevices(podUID, resource string) sets.String {
+	if _, podExists := pdev[podUID]; !podExists {
+		return nil
+	}
+
+	ret := sets.NewString()
+	for _, resources := range pdev[podUID] {
+		for _, devs := range resources {
+			ret = ret.Union(devs.deviceIds)
+		}
+	}
+	return ret
+}
```

Comment:
Could This be simplified by just looping over all containers and Unioning a call to `pdev. containerDevices()`.

Also the current logic doesn't seem to match the description. As it stands, this function returns the set of devices allocated to *all*  resources across the pod (not just the devices for a particular resourcce).

Reply:
Yes, you're right. I simplified function and it is now aligned with the description.